BacktraceScreen.c

```diff
@@ -0,0 +1,527 @@
+/*
+htop - BacktraceScreen.h
+(C) 2025 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h" // IWYU pragma: keep
+
+#include "BacktraceScreen.h"
+
+#if defined(HAVE_BACKTRACE_SCREEN)
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+
+#include "CRT.h"
+#include "FunctionBar.h"
+#include "Macros.h"
+#include "Object.h"
+#include "Panel.h"
+#include "Platform.h"
+#include "Process.h"
+#include "ProvideCurses.h"
+#include "RichString.h"
+#include "Settings.h"
+#include "Vector.h"
+#include "XUtils.h"
+
+
+#define MAX_HEX_ADDR_STR_LEN_32 (strlen("0x") + 8)
+#define MAX_HEX_ADDR_STR_LEN_64 (strlen("0x") + 16)
+
+typedef enum BacktracePanelOptions_ {
+   OPTION_NAME_DEMANGLE,
+   OPTION_NAME_RAW,
+   OPTION_OBJECT_FULL_PATH,
+   OPTION_OBJECT_BASENAME,
+   LAST_PANEL_OPTION,
+} BacktracePanelOptions;
+
+static const char* const BacktracePanel_options[LAST_PANEL_OPTION] = {
+   [OPTION_NAME_DEMANGLE] = "Demangle",
+   [OPTION_NAME_RAW] = "Raw",
+   [OPTION_OBJECT_FULL_PATH] = "Full Path",
+   [OPTION_OBJECT_BASENAME] = "Basename",
+};
+
+typedef enum BacktraceFrameHeaders_ {
+   HEADER_NUMBER_FRAME,
+   HEADER_ADDRESS,
+   HEADER_NAME,
+   HEADER_PATH,
+   LAST_PANEL_HEADER,
+} BacktracePanelHeaders;
+
+static const char* const BacktraceFrame_headerFields[LAST_PANEL_HEADER] = {
+   [HEADER_NUMBER_FRAME] = "#",
+   [HEADER_ADDRESS] = "ADDRESS",
+   [HEADER_NAME] = "NAME",
+   [HEADER_PATH] = "PATH",
+};
+
+static const char* const BacktraceScreenFunctions[] = {
+   "Refresh",
+#if defined(HAVE_DEMANGLING)
+   BacktracePanel_options[OPTION_NAME_RAW],
+#endif
+   BacktracePanel_options[OPTION_OBJECT_FULL_PATH],
+   "Done   ",
+   NULL
+};
+
+static const char* const BacktraceScreenKeys[] = {
+   "F1",
+#if defined(HAVE_DEMANGLING)
+   "F2",
+#endif
+   "F3",
+   "Esc",
+   NULL
+};
+
+static const int BacktraceScreenEvents[] = {
+   KEY_F(1),
+#if defined(HAVE_DEMANGLING)
+   KEY_F(2),
+#endif
+   KEY_F(3),
+   27,
+};
```

Comment:
Please check with function bar implementation. There was some issue with these arrays if too short causing OOB reads when used as a function bar (using a memory sanitizer).

Reply:
I just checked with `fsanitize` when `HAVE_DEMANGLING` is present and when it's not present. It works, No OOB.